src/mbgl/util/tile_coordinate.hpp

```diff
@@ -0,0 +1,39 @@
+#ifndef MBGL_UTIL_TILE_COORDINATE
+#define MBGL_UTIL_TILE_COORDINATE
+
+#include <mbgl/style/types.hpp>
+#include <mbgl/map/transform_state.hpp>
+
+
+namespace mbgl {
+
+class TransformState;
+
+// Has floating point x/y coordinates.
+// Used for computing the tiles that need to be visible in the viewport.
```

Comment:
It's main purpose is represent euclidean world coordinates and help convert them to tile coordinates at different zoom levels. It does this in a weird way though, since it also has a z parameter, which gives it an extra dimension.

It should probably be replaced by a zoom-less world coordinate type with transform methods that convert it to tile units or tile ids.

It was removed in the coordinate cleanup pull request but those changes never made their way to -js. -js uses this class for a couple of things:
- for calculating tile cover
- for helping convert lat/lng to tile coordinates
- in Transform.setLocationAtPoint which is used by easeTo

This class could be removed now and a latlng -> tile units method could be added but I'd rather do this as part of a larger Transform cleanup and synchronization between -js and -native.

Do you see this as a blocker for this pr or as something that can be cleaned up later?